test/EndToEnd/tests/ServicesTest.ps1

```diff
@@ -703,6 +703,40 @@ function Test-BatchEventsApi
     Assert-True $result
 }
 
+function Test-ExecuteInitScriptsPerSolution
```

Comment:
To verify that scripts aren't running multiple times you will need to create a package with a custom init.ps1 script. In the init.ps1 script set a failure env var or some other state if it loads and has already been initialized.

From the test verify that the failure state isn't set. Also verify that the init.ps1 was loaded.

You can look at the other init.ps1 tests for examples of this.